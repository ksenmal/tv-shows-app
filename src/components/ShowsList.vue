<script setup>
import ShowCard from "./ShowCard.vue";
defineProps({
  list: Array,
});
</script>

<template>
  <p v-if="!list || !list.length" class="empty-list-msg">No TV shows found</p>
  <ul v-else class="show-list">
    <li v-for="item in list" :key="item.id" class="list-item">
      <ShowCard :item="item" />
    </li>
  </ul>
</template>

<style scoped>
.show-list {
  padding: 0;
  display: flex;
  flex-wrap: wrap;
  column-gap: 1rem;
  justify-content: center;
}
.list-item {
  list-style: none;
  width: 250px;
  min-height: 320px;
  padding: 20px;
  margin-bottom: 10px;
  animation: fadein 0.5s linear;
  border: 2px solid var(--color-hover);
  border-radius: 4px;
}
@keyframes fadein {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
.list-item img {
  display: inline-block;
  max-width: 100%;
  height: auto;
  vertical-align: middle;
}

.empty-list-msg {
  margin-top: 2rem;
  text-align: center;
}
</style>
