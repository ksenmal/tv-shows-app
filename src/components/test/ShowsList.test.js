import { shallowMount } from "@vue/test-utils";
import ShowsList from "../ShowsList.vue";
import { describe, expect, test } from "vitest";

describe("ShowList suite", () => {
  test("Snapshot when empty", () => {
    const wrapper = shallowMount(ShowsList, {
      props: {
        list: [],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Snapshot with data", () => {
    const wrapper = shallowMount(ShowsList, {
      props: {
        list: [{ id: 1 }, { id: 2 }],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
