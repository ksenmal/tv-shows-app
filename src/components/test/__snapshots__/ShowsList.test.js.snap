// Vitest Snapshot v1

exports[`ShowList suite > Snapshot when empty 1`] = `"<p class=\\"empty-list-msg\\" data-v-f5a7670e=\\"\\">No TV shows found</p>"`;

exports[`ShowList suite > Snapshot with data 1`] = `
"<ul class=\\"show-list\\" data-v-f5a7670e=\\"\\">
  <li class=\\"list-item\\" data-v-f5a7670e=\\"\\">
    <show-card-stub item=\\"[object Object]\\" data-v-f5a7670e=\\"\\"></show-card-stub>
  </li>
  <li class=\\"list-item\\" data-v-f5a7670e=\\"\\">
    <show-card-stub item=\\"[object Object]\\" data-v-f5a7670e=\\"\\"></show-card-stub>
  </li>
</ul>"
`;
